Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM node:4.2.4-wheezy
+MAINTAINER Mark Stemm (mstemm@jut.io)
+
+RUN mkdir -p /opt/juttle-engine
+WORKDIR /opt/juttle-engine
+ADD package.json package.json
+RUN npm install
+COPY . .
```

Comment:
The wisdom of the crowds say this order allows for better caching, as it's more likely that the results of npm install is stable than the code you're changing.